### PR TITLE
Various bug fixes: AP-3320 AP-3324 AP-3325

### PR DESCRIPTION
**Pull request by franklma**
1. AP-3320: Can't copy/cut folder to root directory
https://apromore.atlassian.net/browse/AP-3320

2. AP-3324: Deleting of folder doesn't delete child logs and corresponding XES files
https://apromore.atlassian.net/browse/AP-3324

3. AP-3325: Access rights to parent folders get overridden when sharing artifact
https://apromore.atlassian.net/browse/AP-3325

4. Code formating and clean-up

### CODE REVIEW

File: Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/service/impl/EventLogServiceImpl.java [hunk 1]
```diff
@@ -210,97 +207,100 @@ public SummariesType readLogSummaries(Integer folderId, String searchExpression)
...
-     * @param extension
+     * @param inputStreamLog the inputStream of Log
+     * @param extension      File extension
      * @param domain         the domain of the model
      * @param created        the time created
      * @param publicModel    is this a public model?
-     * @return
-     * @throws Exception
+     * @return Log
+     * @throws Exception UserNotFoundException
```
Comment (raboczi) [inline Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/service/impl/EventLogServiceImpl.java]: I think this needs to be:
```
@throws UserNotFoundException  (explanation)
@throws Exception  (explanation)
```

File: Apromore-Core-Components/Apromore-Manager/src/test/java/org/apromore/test/service/impl/WorkspaceServiceImplTest.java [hunk 2]
```diff
@@ -116,7 +119,9 @@ public final void setUp() throws Exception {
                                                 groupLogRepo,
                                                 logFileService,
                                                 folderServiceImpl,
-                                                storageFacotry,storageRepository);
+                                                storageFacotry,
```
Comment (raboczi) [inline Apromore-Core-Components/Apromore-Manager/src/test/java/org/apromore/test/service/impl/WorkspaceServiceImplTest.java]: storageFacotry -> storageFactory

File: Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/BaseListboxController.java [hunk 3]
```diff
@@ -589,7 +589,7 @@ public void paste() throws Exception {
 			try {
 				copyAndPasteController.paste(targetFolderId);
 			} catch (Exception e) {
-				Messagebox.show("An error is occured during paste process", "Apromore", Messagebox.OK,
+				Messagebox.show("An error is occurred during paste process", "Apromore", Messagebox.OK,
```
Comment (raboczi) [inline Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/BaseListboxController.java]: An error occurred during

**Review by raboczi — COMMENTED** [review]
There's a lot of reformatting here that makes it difficult to pick out the functional changes, but I only spotted superficial things to change.

**Comment by franklma** [comment]
> There's a lot of reformatting here that makes it difficult to pick out the functional changes, but I only spotted superficial things to change.

Right. I will create separate PR for code formatting and clean-up.

**Review by raboczi — APPROVED** [review]
I merged to update with the current development branch so that I was sure it'd build with ApromoreEE.  Everything seems fine.

**Pull request merged**